src/main/java/com/hazelcast/gcp/GcpMetadataApi.java

```diff
@@ -63,7 +68,14 @@ String accessToken() {
     }
 
     private static String extractAccessToken(String accessTokenResponse) {
-        return Json.parse(accessTokenResponse).asObject().get("access_token").asString();
+        try {
+            return Json.parse(accessTokenResponse).asObject().get("access_token").asString();
+        } catch (ParseException e) {
+            LOGGER.severe("Unable to retrieve access token. Please grant permissions to this service account if "
```

Comment:
The same here. All good, but I just wonder if not just to put this message into exception and then use it like this:
```
throw new HazelcastExeception("Unable to retrieve access token. Please grant permissions to this service account if...", e);
```

What do you think? Check [this ](https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern) btw.